numba/np/arrayobj.py

```diff
@@ -2507,11 +2511,25 @@ def array_record_getattr(context, builder, typ, value, attr):
     newdataptr = cgutils.pointer_add(
         builder, array.data, constoffset,  return_type=rary.data.type,
     )
-    datasize = context.get_abi_sizeof(context.get_data_type(dtype))
+    if isinstance(dtype, types.NestedArray):
+        datasize = context.get_abi_sizeof(context.get_data_type(dtype.dtype))
```

Comment:
```suggestion
```
This is calculated again below (after the strides calc)?

Reply:
Whoops, I missed that. Are you happy with this @luk-f-a ?